mlflow/gateway/app.py

```diff
@@ -53,17 +53,37 @@ async def search_routes():
     return {"routes": ACTIVE_ROUTES}
 
 
-def _add_dynamic_route(route: RouteConfig):
-    async def route_endpoint(request: Request):
-        try:
-            # TODO: handle model-specific route logic by mapping the provider to plugin logic
+async def _chat(request: chat.RequestPayload) -> chat.ResponsePayload:
+    return chat.ResponsePayload(**{"candidates": request.messages})
+
 
-            return await request.json()
-        except Exception as e:
-            raise HTTPException(status_code=500, detail=format_exc()) from e
+async def _completions(request: completions.RequestPayload) -> completions.ResponsePayload:
+    return completions.ResponsePayload(**{"candidates": request.messages})
 
+
+async def _embeddings(request: embeddings.RequestPayload) -> embeddings.ResponsePayload:
+    import random
+
+    length = len(request.text)
+    return embeddings.ResponsePayload(
+        **{"embeddings": [random.randint(0, length - 1) / length for _ in range(10)]}
+    )
+
+
+def _route_type_to_endpoint(route_type: RouteType):
+    if route_type == RouteType.Chat:
+        return _chat
+    elif route_type == RouteType.Completions:
+        return _completions
+    # elif route_type == RouteType.Embeddings:
```

Comment:
did we want this to be commented out?

Reply:
I commented this out because we don't have `RouteType.Embeddings` yet

Reply:
Can we uncomment and add it?

Reply:
Uncommented, #8711 will be merged soon.